File: admin-service/internal/models/billing.go
package models

type BillingAccount struct {
	UserID               string `json:"user_id"`
	Email                string `json:"email,omitempty"`
	Nickname             string `json:"nickname,omitempty"`
	AvailableBalanceYuan string `json:"available_balance_yuan"`
	ReservedBalanceYuan  string `json:"reserved_balance_yuan"`
	TotalRechargedYuan   string `json:"total_recharged_yuan"`
	TotalSpentYuan       string `json:"total_spent_yuan"`
	TotalTrafficBytes    int64  `json:"total_traffic_bytes"`
	Status               int32  `json:"status"`
	Version              int32  `json:"version"`
	UpdatedAt            string `json:"updated_at"`
}

type BillingAccountListResponse struct {
	Total    int64            `json:"total"`
	Page     int32            `json:"page"`
	PageSize int32            `json:"page_size"`
	Items    []BillingAccount `json:"items"`
}

type BillingLedgerEntry struct {
	EntryNo                   string `json:"entry_no"`
	UserID                    string `json:"user_id"`
	Email                     string `json:"email,omitempty"`
	Nickname                  string `json:"nickname,omitempty"`
	OrderNo                   string `json:"order_no"`
	HoldNo                    string `json:"hold_no"`
	HistoryID                 int64  `json:"history_id"`
	TaskID                    string `json:"task_id"`
	TransferID                string `json:"transfer_id"`
	OperationID               string `json:"operation_id"`
	EntryType                 int32  `json:"entry_type"`
	Scene                     int32  `json:"scene"`
	ActionAmountYuan          string `json:"action_amount_yuan"`
	AvailableDeltaYuan        string `json:"available_delta_yuan"`
	ReservedDeltaYuan         string `json:"reserved_delta_yuan"`
	BalanceAfterAvailableYuan string `json:"balance_after_available_yuan"`
	BalanceAfterReservedYuan  string `json:"balance_after_reserved_yuan"`
	OperatorUserID            string `json:"operator_user_id"`
	Remark                    string `json:"remark"`
	CreatedAt                 string `json:"created_at"`
}

type BillingLedgerListResponse struct {
	Total    int64                `json:"total"`
	Page     int32                `json:"page"`
	PageSize int32                `json:"page_size"`
	Items    []BillingLedgerEntry `json:"items"`
}

type BillingShortfallOrder struct {
	OrderNo            string `json:"order_no"`
	UserID             string `json:"user_id"`
	Email              string `json:"email,omitempty"`
	Nickname           string `json:"nickname,omitempty"`
	HistoryID          int64  `json:"history_id"`
	TaskID             string `json:"task_id"`
	Scene              int32  `json:"scene"`
	Status             int32  `json:"status"`
	PricingVersion     int32  `json:"pricing_version"`
	ActualIngressBytes int64  `json:"actual_ingress_bytes"`
	ActualEgressBytes  int64  `json:"actual_egress_bytes"`
	ActualTrafficBytes int64  `json:"actual_traffic_bytes"`
	HeldAmountYuan     string `json:"held_amount_yuan"`
	CapturedAmountYuan string `json:"captured_amount_yuan"`
	ReleasedAmountYuan string `json:"released_amount_yuan"`
	ShortfallYuan      string `json:"shortfall_yuan"`
	Remark             string `json:"remark"`
	CreatedAt          string `json:"created_at"`
	UpdatedAt          string `json:"updated_at"`
}

type BillingShortfallListResponse struct {
	Total    int64                   `json:"total"`
	Page     int32                   `json:"page"`
	PageSize int32                   `json:"page_size"`
	Items    []BillingShortfallOrder `json:"items"`
}

type BillingUsageRecord struct {
	UsageNo            string `json:"usage_no"`
	OrderNo            string `json:"order_no"`
	UserID             string `json:"user_id"`
	Email              string `json:"email,omitempty"`
	Nickname           string `json:"nickname,omitempty"`
	HistoryID          int64  `json:"history_id"`
	TaskID             string `json:"task_id"`
	TransferID         string `json:"transfer_id"`
	Direction          int32  `json:"direction"`
	TrafficBytes       int64  `json:"traffic_bytes"`
	UnitPriceYuanPerGB string `json:"unit_price_yuan_per_gb"`
	AmountYuan         string `json:"amount_yuan"`
	PricingVersion     int32  `json:"pricing_version"`
	SourceService      string `json:"source_service"`
	Status             int32  `json:"status"`
	CreatedAt          string `json:"created_at"`
	ConfirmedAt        string `json:"confirmed_at,omitempty"`
}

type BillingUsageListResponse struct {
	Total    int64                `json:"total"`
	Page     int32                `json:"page"`
	PageSize int32                `json:"page_size"`
	Items    []BillingUsageRecord `json:"items"`
}

type BillingPricing struct {
	Version               int32  `json:"version"`
	IngressPriceYuanPerGB string `json:"ingress_price_yuan_per_gb"`
	EgressPriceYuanPerGB  string `json:"egress_price_yuan_per_gb"`
	Enabled               bool   `json:"enabled"`
	Remark                string `json:"remark"`
	UpdatedByUserID       string `json:"updated_by_user_id"`
	EffectiveAt           string `json:"effective_at"`
	CreatedAt             string `json:"created_at"`
}

type WelcomeCreditSettings struct {
	Enabled      bool   `json:"enabled"`
	AmountYuan   string `json:"amount_yuan"`
	CurrencyCode string `json:"currency_code"`
	UpdatedAt    string `json:"updated_at"`
	UpdatedBy    string `json:"updated_by"`
}

File: admin-frontend/types/billing.ts
export interface BillingAccount {
  user_id: string;
  email?: string;
  nickname?: string;
  available_balance_yuan: string;
  reserved_balance_yuan: string;
  total_recharged_yuan: string;
  total_spent_yuan: string;
  total_traffic_bytes: number;
  status: number;
  version: number;
  updated_at: string;
}

export interface BillingAccountListResponse {
  total: number;
  page: number;
  page_size: number;
  items: BillingAccount[];
}

export interface BillingLedgerEntry {
  entry_no: string;
  user_id: string;
  email?: string;
  nickname?: string;
  order_no: string;
  hold_no: string;
  history_id: number;
  task_id: string;
  transfer_id: string;
  operation_id: string;
  entry_type: number;
  scene: number;
  action_amount_yuan: string;
  available_delta_yuan: string;
  reserved_delta_yuan: string;
  balance_after_available_yuan: string;
  balance_after_reserved_yuan: string;
  operator_user_id: string;
  remark: string;
  created_at: string;
}

export interface BillingLedgerListResponse {
  total: number;
  page: number;
  page_size: number;
  items: BillingLedgerEntry[];
}

export interface BillingShortfallOrder {
  order_no: string;
  user_id: string;
  email?: string;
  nickname?: string;
  history_id: number;
  task_id: string;
  scene: number;
  status: number;
  pricing_version: number;
  actual_ingress_bytes: number;
  actual_egress_bytes: number;
  actual_traffic_bytes: number;
  held_amount_yuan: string;
  captured_amount_yuan: string;
  released_amount_yuan: string;
  shortfall_yuan: string;
  remark: string;
  created_at: string;
  updated_at: string;
}

export interface BillingShortfallListResponse {
  total: number;
  page: number;
  page_size: number;
  items: BillingShortfallOrder[];
}

export interface BillingUsageRecord {
  usage_no: string;
  order_no: string;
  user_id: string;
  email?: string;
  nickname?: string;
  history_id: number;
  task_id: string;
  transfer_id: string;
  direction: number;
  traffic_bytes: number;
  unit_price_yuan_per_gb: string;
  amount_yuan: string;
  pricing_version: number;
  source_service: string;
  status: number;
  created_at: string;
  confirmed_at?: string;
}

export interface BillingUsageListResponse {
  total: number;
  page: number;
  page_size: number;
  items: BillingUsageRecord[];
}

export interface BillingPricing {
  version: number;
  ingress_price_yuan_per_gb: string;
  egress_price_yuan_per_gb: string;
  enabled: boolean;
  remark: string;
  updated_by_user_id: string;
  effective_at: string;
  created_at: string;
}

export interface BillingBalanceAdjustmentPayload {
  operation_id?: string;
  amount_yuan: string;
  remark: string;
}

export interface BillingPricingUpdatePayload {
  ingress_price_yuan_per_gb: string;
  egress_price_yuan_per_gb: string;
  remark?: string;
}

export interface WelcomeCreditSettings {
  enabled: boolean;
  amount_yuan: string;
  currency_code?: string;
  created_at?: string;
  updated_at?: string;
  updated_by?: string;
}

export interface WelcomeCreditSettingsUpdatePayload {
  enabled: boolean;
  amount_yuan: string;
  currency_code?: string;
}

export interface BillingShortfallReconcilePayload {
  remark?: string;
}
